btc-withdrawal/src/main/kotlin/com/d3/btc/withdrawal/service/BtcRollbackService.kt

```diff
@@ -26,29 +24,30 @@ class BtcRollbackService(
     private val withdrawalConsumer: MultiSigIrohaConsumer
 ) {
 
+    private val rollbackService = RollbackService(withdrawalConsumer)
+
     /**
      * Rollbacks given amount of money to a particular Iroha account
      * @param withdrawalDetails - details of withdrawal to rollback
      * @param reason - reason of rollback
      */
     fun rollback(withdrawalDetails: WithdrawalDetails, reason: String) {
-        val amountToRollbackSat = withdrawalDetails.amountSat + withdrawalDetails.withdrawalFeeSat
-        val rollbackMessage = "$ROLLBACK_DESCRIPTION. $reason"
-        withdrawalConsumer.getConsumerQuorum().flatMap { quorum ->
-            ModelUtil.transferAssetIroha(
-                withdrawalConsumer,
-                withdrawalConsumer.creator,
-                withdrawalDetails.sourceAccountId,
-                BTC_ASSET,
-                rollbackMessage.take(64).toLowerCase(),
-                satToBtc(amountToRollbackSat).toPlainString(),
-                withdrawalDetails.withdrawalTime,
-                quorum
-            )
-        }.fold(
+        val withdrawalFinalizationDetails = WithdrawalFinalizationDetails(
+            satToBtc(withdrawalDetails.amountSat),
+            BTC_ASSET,
+            satToBtc(withdrawalDetails.withdrawalFeeSat),
+            BTC_ASSET,
+            withdrawalDetails.sourceAccountId,
+            withdrawalDetails.withdrawalTime,
+            withdrawalConsumer.creator
```

Comment:
This value stands for `destinationAddress`. Not creator.

Reply:
toAddress is a BTC address here, so only one option to get withdrawal service client id is to get it from consumer